src/lib/player-state-pinger.js

```diff
@@ -16,10 +16,10 @@ class PlayerStatePinger {
 
   setPlayerState(playerId, state, value) {
     this.setState(prevState => {
-      const nextState = Object.assign({}, prevState);
-      const { me, team } = nextState;
-      const player = [me, ...team].find(client => client && client.id === playerId);
-      player[state] = value;
+      const nextState = { ...prevState, playerAction: { ...prevState.playerAction } };
+      if (nextState.playerAction[playerId] === undefined)
+        nextState.playerAction[playerId] = {};
+      nextState.playerAction[playerId][state] = value;
```

Comment:
Since `playerAction` is an object, we can return an new object directly.

```javascript
return {
  ...prevState,
  playerAction: {
    ...prevState.playerAction,
    [playerId]: {
      [state]: value
    }
  }
};
```